cpp/src/arrow/flight/serialization_internal.cc

```diff
@@ -201,9 +193,7 @@ grpc::Status FlightDataSerialize(const FlightPayload& msg, ByteBuffer* out,
   // Write the descriptor if present
   int32_t descriptor_size = 0;
   if (msg.descriptor != nullptr) {
-    if (msg.descriptor->size() > kInt32Max) {
-      return ToGrpcStatus(Status::CapacityError("Descriptor size overflow (>= 2**31)"));
```

Comment:
From the jira issue, looks grpc should check and handle our return value decently.
Maybe open an issue in grpc community?

Reply:
I'll try to get a repro up when I can and file a bug, good idea.

Reply:
See https://github.com/grpc/grpc/issues/26695